### PR TITLE
accounts: bump timeout in TestSignRace

### PR DESCRIPTION
**Pull request by fjl**
This should fix the flakeyness.

### CODE REVIEW

**Comment by obscuren** [comment]
:+1:

**Comment by karalabe** [comment]
:-1: 

```
$ cat accounts_test.go | grep "end := time"
	end := time.Now().Add(500 * time.Millisecond)
```

```
$ godep go test -v -cpu=1,2,4,8 -run SignRace
=== RUN   TestSignRace
--- FAIL: TestSignRace (0.51s)
	accounts_test.go:140: Account did not lock within the timeout
=== RUN   TestSignRace
--- PASS: TestSignRace (0.02s)
=== RUN   TestSignRace
--- PASS: TestSignRace (0.02s)
=== RUN   TestSignRace
--- PASS: TestSignRace (0.02s)
```

File: accounts/accounts_test.go [hunk 1]
```diff
@@ -128,7 +128,7 @@ func TestSignRace(t *testing.T) {
 	if err := am.TimedUnlock(a1.Address, "", 15*time.Millisecond); err != nil {
 		t.Fatalf("could not unlock the test account", err)
 	}
-	end := time.Now().Add(80 * time.Millisecond)
+	end := time.Now().Add(500 * time.Millisecond)
 	for time.Now().Before(end) {
```
Comment (karalabe) [inline accounts/accounts_test.go]: This goroutine is spinning non stop and will not yield the scheduler to the signer. Add `runtime.Gosched()` before repeating the loop. It will fix the failing test if `-cpu=1`

Comment (fjl) [inline accounts/accounts_test.go]: It should yield anyway. See http://golang.org/doc/go1.2#preemption.
I cannot reproduce this failure. Does inserting the call to `Gosched` fix the test for you?

**Comment by fjl** [comment]
@karalabe I've added a call to `time.Sleep` in the loop. Please try now.

**Comment by fjl** [comment]
@karalabe PTAL

**Comment by obscuren** [comment]
Merging this PR. It's been open for too long and the tests work fine. If the test needs to be improved for some corner case, please update in a future PR.

File: accounts/accounts_test.go (hunk 1 above)
Comment (karalabe) [inline accounts/accounts_test.go]: I switched over to Go 1.5, maybe it has a bit different inlining strategy and caused the `Sign` to be inlined and thus not peemted. Dunno really, but the Sleep fix should also be equally good. (Sorry for the late reply, wasn't available when you asked and since it got merged before I got to reply, I figured it can wait, me not having to type on a mobile with shitty cell data access)
